Haptic Feedback

guitaripod opened the pull request:
Adds Haptics to supported devices.

Selection haptics for button presses and when closing views.

Notification haptics for successful events and errors.

Haptics improves accessibility and should be included in an app meant for everyone.

guitaripod committed:
Add HapticEngine

guitaripod committed:
Add Haptic Feedback to Various Buttons and Events

guitaripod committed:
Update CHANGELOG.md

repomies commented:
One place where I really felt the haptic feedback missing was when accepting the terms and voluntary use checkboxes in the onboarding. There it would be good to trigger `selectionChanged`.

guitaripod committed:
Add Haptics to Onboarding

repomies commented on Koronavilkku/Koronavilkku/ViewController/MainViewController.swift:
```diff
@@ -208,10 +208,12 @@ class MainViewController: UIViewController {
     }
     
     @objc private func logoImageTapped() {
+        HapticEngine.result.notificationOccurred(.success)
```
I think playing the `.success` fanfare for just selecting the logo and opening THL's website is a bit too much maybe 😄

And generally I might suggest to keep the haptic feedback only for clear selection, success, failure, and warning situations. Taking a cue from how Apple does it, haptic feedback is never triggered e.g. from simple navigation actions.

guitaripod replied:
Fair enough. Will remove that one.

guitaripod committed:
Remove Haptic From Logo

Commit:
Make some adjustments to the proposed change

- Prevent overuse of haptics by using it only where it makes sense
- Remove the shared HapticEngine object that prevents the engine from being set to idle state

Commit:
Update CHANGELOG.md

Commit:
Update CHANGELOG.md
